File: src/gatling/scala/mq/utils/parser.scala
package  mq.utils

import java.util.UUID.randomUUID
import java.text.SimpleDateFormat
import java.util.Date
import scala.util.Random
import mq.config._;

object Parser  extends  SimulationTraits {

        
    def replacer :Map[String, String] =
        Map[String, String] (
            "_UUID_" -> UUID,
            "_RAND_STR6_" -> randStr(6),
            "_RAND_STR8_" -> randStr(8),
            "_CURRENT_DT_ISO_" -> currentISODate,
            "_TS_" -> unixtimestamp,
            "_RUNID_" -> Config.RUNID,
            "_RAND_INT4_" -> randInt(4),
            "_RAND_INT6_" -> randInt(6),
            "_RAND_INT8_" -> randInt(8),
            "_DATE_" -> todayDate,
            "_TIME_" -> timeNow
        )


    def randStr(n:Int) = Random.alphanumeric.take(n).mkString.toUpperCase()
    
    def randInt(n: Int) = Random.alphanumeric.filter(_.isDigit).take(n).mkString
    
    def r = new scala.util.Random
    
    def roundUp (d: Double) = math.round(d).toInt
    
    def UUID =  randomUUID().toString().replace("-","").toUpperCase()

    def currentISODate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss").format(new Date())

    def unixtimestamp = (System.currentTimeMillis() / 1000L).toString();

    def todayDate = new SimpleDateFormat("yyyy-MM-dd").format(new Date())

    def timeNow = new SimpleDateFormat("HH:mm:ss").format(new Date())

}

File: src/main/java/utils/Parser.java
package utils;
import java.util.UUID;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Random;
import java.util.Map;
import java.util.HashMap;

public class Parser {


    public static String uuidgen() {
        UUID uuid = UUID.randomUUID();
        String uuidAsString = uuid.toString();
        return uuidAsString.replace("-","").toUpperCase();
    }


    public static String  currentISODate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss").format(new Date());

    public static String unixtimestamp = Long.toString((System.currentTimeMillis() / 1000L));

    public static String todayDate = new SimpleDateFormat("yyyy-MM-dd").format(new Date());

    public static String timeNow = new SimpleDateFormat("HH:mm:ss").format(new Date());

    public static String randStr(Integer targetStringLength)  {
        int leftLimit = 48; // numeral '0'
        int rightLimit = 122; // letter 'z'
        Random random = new Random();
    
        String generatedString = random.ints(leftLimit, rightLimit + 1)
          .filter(i -> (i <= 57 || i >= 65) && (i <= 90 || i >= 97))
          .limit(targetStringLength)
          .collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append)
          .toString();

          return generatedString;
    }

    public static String randInt(Integer targetStringLength)  {
        int leftLimit = 48; // numeral '0'
        int rightLimit = 57; // numeral '10'
        Random random = new Random();
    
        String generatedString = random.ints(leftLimit, rightLimit + 1)
          .filter(i -> (i <= 57 || i >= 65) && (i <= 90 || i >= 97))
          .limit(targetStringLength)
          .collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append)
          .toString();

          return generatedString;
    }


    public static  Map<String, String> replacerMap = new HashMap<String, String>()
        {
            {
                put("UUID", uuidgen());
                put("RAND_STR6", randStr(6));
                put("RAND_STR8", randStr(8));
                put("CURRENT_DT_ISO", currentISODate);
                put("TS", unixtimestamp);
                put("RUNID", "1234");
                put("RAND_INT4", randInt(4));
                put("RAND_INT6", randInt(6));
                put("RAND_INT8", randInt(8));
                put("DATE", todayDate);
                put("TIME", timeNow);

            };
        };

}
